### PR TITLE
creating 3_keras_sequential_api notebook for review

### PR DESCRIPTION
**Pull request by munnm**
finished 3_keras_sequential_api notebook for review

### CODE REVIEW

File: courses/machine_learning/intro_to_tensorflow/3_keras_sequential_api.ipynb [hunk 1]
```diff
@@ -0,0 +1,482 @@
+{
```
Comment (BenoitDherin) [inline courses/machine_learning/intro_to_tensorflow/3_keras_sequential_api.ipynb]: <p>line 9: "create" -&gt; "Create"</p><p>line 13: I'd remove the spaces between the beginning of the line and "}"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/559/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (BenoitDherin) [inline courses/machine_learning/intro_to_tensorflow/3_keras_sequential_api.ipynb]: <p>To have very simple &amp; readable model code, I'd import all the needed layers and models at the beginning where all the other imports are made:</p><p> </p><p> </p><p><code>from tensorflow.kears.models import Sequential</code></p><p><code>from tensorflow.keras.layers import (</code></p><p><code>    DenseFeature,</code></p><p><code>    Dense,</code></p><p><code>)</code></p><p> </p><p> </p><p>Then I'd define the model directly by passing the list of layers to the "Sequential" ; this would look like the following:</p><p> </p><p><code>model = Sequential([</code></p><p><code>    DenseFeatures(feature_columns=feature_columns.values()),</code></p><p><code>    Dense(units=32, activation="relu", name="h1") ,</code></p><p><code>    Dense(units=8, activation="relu", name="h2"),</code></p><p><code>    Dense(units=1, activation="linear", name="output")</code></p><p><code>])</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/559/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (BenoitDherin) [inline courses/machine_learning/intro_to_tensorflow/3_keras_sequential_api.ipynb]: <p>It seems that the model.fit method inspects the data, and if it's a generator will invoke automatically .fit_generator, so we don't really need to use it directly: .fit is enough even in the case we pass a tf.data.Dataset</p><p> </p><p>(It's here in the Keras code: <a href="https://github.com/keras-team/keras/blob/master/keras/engine/training.py#L1130-L1147" target="_blank">https://github.com/keras-team/keras/blob/master/keras/engine/training.py#L1130-L1147</a>)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/559/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (BenoitDherin) [inline courses/machine_learning/intro_to_tensorflow/3_keras_sequential_api.ipynb]: <p>If we set</p><p> </p><p>NUM_EVALS = 50</p><p>TRAIN_BATCH_SIZE = 1000</p><p> </p><p>then the rmse curves look much better (already with NUM_EVAL 10)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/559/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (BenoitDherin) [inline courses/machine_learning/intro_to_tensorflow/3_keras_sequential_api.ipynb]: <p>To be able to inspect the model in Tensorboard, I'd add:</p><p> </p><p>at the beginning: </p><p>from tensorflow.keras.callbacks import TensorBoard</p><p> </p><p>then here I'd pass "[TensorBoard(LOGDIR)]" to the callbacks argument of the fit method. </p><p> </p><p>Then we can launch tensorboard on the LOGDIR.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/559/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (BenoitDherin) [inline courses/machine_learning/intro_to_tensorflow/3_keras_sequential_api.ipynb]: <p>A simpler way to do that would be</p><p> </p><p>RMSE_COLS = ['rmse', 'val_rmse']</p><p>pd.DataFrame(history.history)[RMSE_COLS].plot();</p><p> </p><p>(same thing for the loss plot)</p><p> </p><p>For the above to work, I believe you'd need to do "%matplotlib inline" (a good place would be just after the imports)</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/559/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Comment (munnm) [reply courses/machine_learning/intro_to_tensorflow/3_keras_sequential_api.ipynb]: Ok. Sounds goood.

Comment (munnm) [reply courses/machine_learning/intro_to_tensorflow/3_keras_sequential_api.ipynb]: will do

Comment (munnm) [reply courses/machine_learning/intro_to_tensorflow/3_keras_sequential_api.ipynb]: ok. interesting. I was getting an odd error when I used just .fit. Googling/searching didn't help make must sense of it. I'll show you in our sync.

Comment (munnm) [reply courses/machine_learning/intro_to_tensorflow/3_keras_sequential_api.ipynb]: ok thanks!
